packages/in_app_purchase/in_app_purchase_storekit/ios/Classes/FIAObjectTranslator.m

```diff
@@ -277,7 +277,7 @@ + (SKPaymentDiscount *)getSKPaymentDiscountFromMap:(NSDictionary *)map
     return nil;
   }
 
-  if (!timestamp || ![timestamp isKindOfClass:NSNumber.class] || [timestamp intValue] <= 0) {
+  if (!timestamp || ![timestamp isKindOfClass:NSNumber.class] || [timestamp longLongValue] <= 0) {
```

Comment:
Nice catch. Can we add an XCUnitTest for this? Maybe have the timestamp to be a very big long long number?

Reply:
Done. I've added XCUnitTest using current timestamp number value. It's enough to cause the `intValue` overflow in negative number. If I use really big long long number, the intValue overflow goes positive again and passes the test. I'm not sure of the usefulness of this "not negative int" check in the original code.